# 37) Binom Dağılımı
Binom dağılımı, bağımsız ve iki olası sonucu (başarı/başarısızlık) olan n sayıda denemedeki toplam başarı sayısını veren ayrık bir olasılık dağılımıdır. 

## Binom Dağılımının Temel Özellikleri
1. İki sonuç: Her deneme sadece iki sonuç verir (örneğin yazı-tura, evet-hayır).
2. Bağımsızlık: Denemeler birbirini etkilemez.
3. Sabit olasılık: Her denemede başarı olasılığı (p) aynıdır

## Bernoulli Dağılımı ve Binom Dağılımı Arasındaki Fark
Bernoulli dağılımı sadece tek bir denemenin sonucunu modellerken, binom dağılımı bu denemenin ardışık ve bağımsız olarak **n** kere tekrarlanmasıyla elde edilen toplam başarı sayısını modeller.

## Parametreler ve Formül
Binom dağılımının iki parametresi vardır:
- **n:** Toplam deneme sayısı
- **p:** Her denemedeki başarı olasılığı

**Olasılık Kütle Fonksiyonu (PMF):**
$$P(X=k) = \binom{n}{k} p^k (1-p)^{n-k}$$

Burada $\binom{n}{k}$ (kombinasyon), n denemeden hangi k tanesinin başarı olacağının kaç farklı şekilde gerçekleşebileceğini sayar örneğin 10 denemeden tam 3'ünün başarı olması, farklı 3'lü kombinasyonlarla gerçekleşebilir (1-2-3. denemeler başarılı olabilir, ya da 4-7-9. denemeler olabilir, vs.), bu formül hepsini hesaba katar.

## Beklenen Değer ve Varyans
$$E(X) = np \qquad Var(X) = np(1-p)$$

Dikkat edersek, bu formüller Bernoulli'nin $E(X)=p$, $Var(X)=p(1-p)$ formüllerinin **n katı** gibi görünüyor bu tesadüf değil, çünkü Binom zaten n tane bağımsız Bernoulli denemesinin toplamıdır.

## Kullanım Koşulları (Ne Zaman Binom Kullanılır?)
1. Sabit sayıda deneme (n) olmalı
2. Her deneme sadece 2 sonuçlu olmalı
3. Denemeler birbirinden bağımsız olmalı
4. Başarı olasılığı (p) her denemede sabit kalmalı

In [ ]:
import numpy as np
from scipy import stats

deneme_sayisi = 10
p = 0.5
binom_beklenen_deger = deneme_sayisi * p
binom_varyans = (deneme_sayisi * p) * (1-p)

print(f"Teorik Beklenen Değer E(X): {binom_beklenen_deger}")
print(f"Teorik Varyans Var(X): {binom_varyans}")

# --- Simülasyon (Deneysel Değerler) ---
# scipy'nin bernoulli.rvs() fonksiyonu, p olasılıklı 10000 tane 
# rastgele Bernoulli denemesi (0 veya 1) üretir
np.random.seed(42)
denemeler = stats.binom.rvs(n=deneme_sayisi, p=p, size=10000)

deneysel_ortalama = np.mean(denemeler)
deneysel_varyans = np.var(denemeler)

print(f"\nDeneysel Ortalama (10000 deneme): {deneysel_ortalama}")
print(f"Deneysel Varyans (10000 deneme): {deneysel_varyans}")

# --- Karşılaştırma ---
print(f"\nOrtalama farkı: {abs(binom_beklenen_deger - deneysel_ortalama):.4f}")
print(f"Varyans farkı: {abs(binom_varyans - deneysel_varyans):.4f}")

degerler, sayilar = np.unique(denemeler, return_counts=True) # return_counts dizideki her bir değerin frekansını veriyor.
for d, s in zip(degerler, sayilar):
    print(f"{d} başarı: {s} kere ({s/10000*100:.1f}%)")

Teorik Beklenen Değer E(X): 5.0
Teorik Varyans Var(X): 2.5

Deneysel Ortalama (10000 deneme): 4.9684
Deneysel Varyans (10000 deneme): 2.46720144

Ortalama farkı: 0.0316
Varyans farkı: 0.0328
0 başarı: 16 kere (0.2%)
1 başarı: 85 kere (0.9%)
2 başarı: 451 kere (4.5%)
3 başarı: 1215 kere (12.2%)
4 başarı: 2055 kere (20.5%)
5 başarı: 2509 kere (25.1%)
6 başarı: 2021 kere (20.2%)
7 başarı: 1121 kere (11.2%)
8 başarı: 436 kere (4.4%)
9 başarı: 84 kere (0.8%)
10 başarı: 7 kere (0.1%)


### Sonuç
Yukarıdaki öncelikle formül üzerinden hazırladığımız beklenen değer ve varyans değerleri ile kendi simülasyonumuz (10 kere madeni para atışını 10000 kere tekrarlama) sonucu elde ettiğimiz beklenen değer ve varyans değerlerini karşılaştırdığımızda aradaki farkların sırasıyla 0.0316 ve 0.0328 olduğunu görüyoruz. Bu atış simülasyonunu ne kadar çok kez tekrarlarsak aradaki farkın daha da 0'a yaklaşacağını söyleyebiliriz.